### PR TITLE
Create Linear Regression classes

### PR DESCRIPTION
**Pull request by Squadula**
- [x] Linear Regression base class
- [x] Chaospy Linear Regression class
    - supported models:
        - [x] Monomials
        - [x] Hermite polynomials
        - [x] Legendre polynomials
- [x] Jupyter Notebook as API example: `examples/fit/surrogate.ipynb` (GP and LinReg)
- [x] integration test (mockup)
- [x] restructured Surrogate base class
    - moved `pre_train`, `post_train` and `pre_predict` from `GaussianProcess` to `Surrogate`
- [x] polishing for `v0.5`
    - clarified docs for *Variables*
    - added draft for new *Components*
    - set up `pre-commit` (`black` and `nbconvert` to clean notebooks, preparation for `v0.6`)
    - updated *Contributing*
---
**ToDo** in a later PR:
- Sklearn & Custom Linear Regression class ?
- Fourier / Gaussian / Sigmoid basis transformers ? 
- Bayesian Ridge ?
- `optimize` hyperparameters (`sigma_n` and `sigma_p`)
- Multi Output Regression
- unit tests
- documentation (& add example notebook to docs)
---
<details><summary>Linear Regression with GORILLA data</summary>

image

→ it fits the data very well but can't model the ridge
</details>

### CODE REVIEW

**Review by Rykath — CHANGES_REQUESTED** [review]
`custom_linreg.py` and `sklearn_linreg.py` are not complete or tested yet. They should be disabled or moved to a development branch.

`tests/sur/*` are manual tests with large sections of commented code, they should be moved to a development branch.
